### PR TITLE
Proposal: All the non-Node-specific fileSystem methods to allow both strings and Uris

### PR DESCRIPTION
**Pull request by sadasant**
How about this change? It would help me streamline some of the web Variable Viewer changes.

Pros:
- Centralizes new functionalities without requiring changes at the leaf references of these methods.
- Allows new features to be implemented on top of URIs without forcing old features to be updated (reducing the number of possible bugs that could arise from changing things that are already working).

Cons:
- Makes these file system methods and API a bit more complex.

### CODE REVIEW

**Review by rchiodo — CHANGES_REQUESTED** [review]
I guess I don't like this idea. I'd rather just make everything that uses this class use URIs. This feels like it's doing too much to me.

The two functions - localDirectoryExists and localFileExists force strings because 'local' files have to support fsPath. It reenforces this fact.

Making everything support strings and URIs muddies this enforcement. It's now unclear what you should pass to any of the FileSystem apis. Is fsPath okay? Sure because it will accept it. 

But it isn't. We want all paths to be URIs if we can. Cause then there's no errors with using fsPath by mistake.

File: src/platform/common/platform/fileSystem.ts [hunk 1]
```diff
@@ -15,6 +15,20 @@ import { traceError } from '../../logging';
 
 const ENCODING = 'utf8';
 
+function ensureUri(path: string | vscode.Uri): vscode.Uri {
```
Comment (DonJayamanne) [inline src/platform/common/platform/fileSystem.ts:18]: I think you can use the existing `isUri` function instead of these two.

**Review by DonJayamanne — CHANGES_REQUESTED** [review]
I don't think we should mix and match strings with Uris, 
We've found that always using Uris is the best and thats what we've decided to go with as that avoids a lot of confusions, e.g. I can see someone passing a local file path as a string and this will fail as `Uri.parse` will fail as the string passed is not a uri value.

Also the use of Uri simple avoids such issues during complication.

File: src/platform/common/platform/fileSystem.ts [hunk 2]
```diff
@@ -38,76 +54,76 @@ export class FileSystem implements IFileSystem {
...
 
-    async createDirectory(uri: vscode.Uri): Promise<void> {
-        await this.vscfs.createDirectory(uri);
+    async createDirectory(uri: string | vscode.Uri): Promise<void> {
+        await this.vscfs.createDirectory(ensureUri(uri));
     }
 
-    async delete(uri: vscode.Uri): Promise<void> {
-        await this.vscfs.delete(uri);
+    async delete(uri: string | vscode.Uri): Promise<void> {
```
[the remark this answers is not in the document]
Comment (rchiodo) [reply src/platform/common/platform/fileSystem.ts:110]: See my comment :)

I think Daniel is closing this one. His newer PR is a better way to address this.

**Pull request closed without merging**